Source/DafnyCore/AST/Statements/Statements.cs

```diff
@@ -245,7 +245,19 @@ public ProduceStmt(IToken tok, IToken endTok, List<AssignmentRhs> rhss)
       foreach (var e in base.NonSpecificationSubExpressions) { yield return e; }
       if (Rhss != null) {
         foreach (var rhs in Rhss) {
-          foreach (var ee in rhs.SubExpressions) {
+          foreach (var ee in rhs.NonSpecificationSubExpressions) {
+            yield return ee;
+          }
+        }
+      }
+    }
+  }
+  public override IEnumerable<Expression> SpecificationSubExpressions {
+    get {
+      foreach (var e in base.SpecificationSubExpressions) { yield return e; }
+      if (Rhss != null) {
+        foreach (var rhs in Rhss) {
```

Comment:
I like to write this sort of code like that:
```suggestion
        foreach (var rhs in Rhss ?? Enumerable.Empty<…>) {
```

Reply:
Changed.